File: client/src/app/modules/quotation/services/assignation.type.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { AssignationType } from '../../quotation/model/AssignationType';

@Injectable({
  providedIn: 'root'
})
export class AssignationTypeService extends AppRestService<AssignationType>{

  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  getAssignationTypes() {
    return this.getList(new HttpParams(), "assignation-types");
  }
}

File: src/main/java/com/jss/osiris/modules/osiris/quotation/service/AssignationTypeService.java
package com.jss.osiris.modules.osiris.quotation.service;

import java.util.List;

import com.jss.osiris.modules.osiris.quotation.model.AssignationType;

public interface AssignationTypeService {
    public List<AssignationType> getAssignationTypes();

    public AssignationType getAssignationType(Integer id);

    public AssignationType addOrUpdateAssignationType(AssignationType assignationType);

    public void updateAssignationTypes();
}
